src/app/challenges/[id]/ChallengeEditor.tsx

```diff
@@ -37,10 +37,22 @@ export default function ChallengeEditor({
   }, [snack]);
 
   const { files } = snackState;
+  const challengesWithUserDefinedStrings = [2, 4]
```

Comment:
This is part of the challenge, so I would make it a property on the challenge type that defaults to false and gets used here instead. Otherwise someone creating a new challenge (unlikely for the foreseeable future but humour me) might be more likely to miss this

Reply:
Yeah, good point. I'm not sure why I didn't think to put it as a property on the challenge but have updated now